pkg/stub/handler.go

```diff
@@ -268,6 +274,62 @@ func (h *Handler) ensureCoreDNSForClusterDNS(dns *dnsv1alpha1.ClusterDNS) error
 	return nil
 }
 
+// ensureExternalNameForOpenshiftService ensures 'openshift.default.svc'
+// resolves to 'kubernetes.default.svc'.
+// This will ensure backward compatibility with openshift 3.x
+func (h *Handler) ensureExternalNameForOpenshiftService() error {
+	svc := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "openshift",
+			Namespace: "default",
+		},
+		Spec: corev1.ServiceSpec{
+			Type:         corev1.ServiceTypeExternalName,
+			ExternalName: "kubernetes.default.svc.cluster.local",
+			Ports: []corev1.ServicePort{{
```

Comment:
Is `Ports` actually required here? I don't have a cluster up yet to test, but if memory serves when I tested this solution only `Type` and `ExternalName` were required to get the CNAME.

Reply:
Did not find enough documentation on ExternalName service type. Tested accessing openshift service on port 443 without adding port entries in the openshift service spec and it is working as expected. So, ports field is not required here.